app/javascript/components/App.js

```diff
@@ -1,13 +1,15 @@
 import React from 'react';
-import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
 import Greeting from './Greeting';
 import { GreetingProvider } from '../contexts/GreetingContext';
 
 function App() {
   return (
     <GreetingProvider>
       <Router>
-        <Route path="/greeting" component={Greeting} />
+        <Routes>
+          <Route path="/greeting" element={<Greeting />} />
+        </Routes>
       </Router>
     </GreetingProvider>
```

Comment:
- [ ] Great work creating the store, however, The random messages are not retrieved from the store before they are displayed. They are fetched directly using the `useEffect` hook in the `Greetings` component. As highlighted by the previous reviewer, you need the redux store for state management. When you dispatch an action that fetches the random greeting, it should be stored in the redux store and accessed by te React component using the `useSelector` hook. Kindly implement this. 👍 😎